app/scripts/background.js

```diff
@@ -825,37 +827,113 @@ export function setupController(
     updateBadge,
   );
 
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.METAMASK_NOTIFICATIONS_LIST_UPDATED,
+    updateBadge,
+  );
+
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.METAMASK_NOTIFICATIONS_MARK_AS_READ,
+    updateBadge,
+  );
+
+  controller.controllerMessenger.subscribe(
+    METAMASK_CONTROLLER_EVENTS.NOTIFICATIONS_STATE_CHANGE,
+    updateBadge,
+  );
+
   controller.txController.initApprovals();
 
   /**
    * Updates the Web Extension's "badge" number, on the little fox in the toolbar.
    * The number reflects the current number of pending transactions or message signatures needing user approval.
    */
   function updateBadge() {
+    const unapprovedTransactionCount = getUnapprovedTransactionCount();
+    const unreadNotificationsCount = getUnreadNotificationsCount();
+
     let label = '';
-    const count = getUnapprovedTransactionCount();
-    if (count) {
-      label = String(count);
+    // eslint-disable-next-line @metamask/design-tokens/color-no-hex
+    let badgeColor = '#0376C9';
+    if (unapprovedTransactionCount) {
+      label = '\u22EF'; // unicode ellipsis
+    } else if (unreadNotificationsCount > 0) {
+      label =
+        unreadNotificationsCount > 9
+          ? String('9+')
```

Comment:
We shouldn't need to cast this since it's already a string.

Also, rather than coupling the label to a specific value, could we instead define a constant such as `BADGE_MAX_NOTIFICATION_COUNT` and then dynamically generate the string like `${BADGE_MAX_NOTIFICATION_COUNT}+`?